discord-bot/bot/extensions/tasks.py

```diff
@@ -0,0 +1,304 @@
+# -*- coding: utf-8 -*-
+"""Task plugin for testing different data collection methods."""
+import asyncio
+import logging
+import typing as t
+from datetime import datetime, timedelta
+
+import hikari
+import lightbulb
+import lightbulb.decorators
+import miru
+from oasst_shared.schemas.protocol import TaskRequestType
+
+from bot.utils import format_time
+
+plugin = lightbulb.Plugin("TaskPlugin")
+
+MAX_TASK_TIME = 60 * 60
+MAX_TASK_ACCEPT_TIME = 60
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+@plugin.command
+@lightbulb.option(
+    "type",
+    "The type of task to request.",
+    choices=[hikari.CommandChoice(name=task.split(".")[-1], value=task) for task in TaskRequestType],
+    required=False,
+    default=TaskRequestType.summarize_story,
+    type=str,
+)
+@lightbulb.command("task_thread", "Request a task from the backend.", ephemeral=True)
+@lightbulb.implements(lightbulb.SlashCommand)
+async def task_thread(ctx: lightbulb.SlashContext):
+    """Request a task from the backend."""
+    typ: str = ctx.options.type
+
+    # Create a thread for the task
+    thread = await ctx.bot.rest.create_thread(ctx.channel_id, hikari.ChannelType.GUILD_PUBLIC_THREAD, f"Task: {typ}")
+
+    await ctx.respond(f"Please complete the task in the thread: {thread.mention}")
+
+    # Send the task in the thread
+    await thread.send(
+        f"""\
+Please complete the task.
+Sample Task
+
+Self destruct {format_time(datetime.now() + timedelta(seconds=MAX_TASK_TIME), 'R')}
+"""
+    )
+
+    # Wait for the user to respond
+    try:
+        event = await ctx.bot.wait_for(
+            hikari.GuildMessageCreateEvent,
+            timeout=MAX_TASK_TIME,
+            predicate=lambda e: e.author.id == ctx.author.id and e.channel_id == thread.id,
+        )
+        await ctx.respond(f"Received message: {event.message.content}")
+    except asyncio.TimeoutError:
+        await ctx.respond("You took too long to respond.")
+    finally:
+        await thread.delete()
+
+
+@plugin.command
+@lightbulb.option(
+    "type",
+    "The type of task to request.",
+    choices=[hikari.CommandChoice(name=task.split(".")[-1], value=task) for task in TaskRequestType],
+    required=False,
+    default=TaskRequestType.summarize_story,
+    type=str,
+)
+@lightbulb.command("task_dm", "Request a task from the backend.", ephemeral=True)
+@lightbulb.implements(lightbulb.SlashCommand, lightbulb.PrefixCommand)
+async def task_dm(ctx: lightbulb.Context):
+    """Request a task from the backend."""
+    await ctx.respond("Please complete the task in your DMs")
```

Comment:
these hard-coded message strings should ideally be moved outside the main code files for people to edit.